src/containers/GraphSettingsContainer.tsx

```diff
@@ -181,7 +183,7 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps> {
     const displaySettingItems = visibilityLayers.map((item: VisibilityLayersType) => (
       <div id={item.id} key={item.id}>
         <label>
-          <input type="checkbox" checked={item.value} onChange={() => item.onChange()} />
+          <input type="checkbox" checked={item.value} onChange={() => item.onChange()} disabled={item.disabled} />
           <span className={checkboxStyle}>{item.labelText}</span>
```

Comment:
To gray out, you would have to create a new style and apply to this element when the item is disabled:

```tsx
const disabledTextStyle = style({color: 'gray'}); // or PFColors.Gray
// ...
<span className={checkboxStyle item.disabled? disabledTextStyle : ''}>{item.labelText}</span>
```